bql/semantic/expression.go

```diff
@@ -175,27 +175,26 @@ func (e *comparisonForLiteral) Evaluate(r table.Row) (bool, error) {
 	if err != nil {
 		return false, fmt.Errorf("in comparisonForLiteral.Evaluate got error: %v", err)
 	}
-
-	var (
-		csEL, csER string
-	)
-
-	formatLiteral := func(lit string) (string, error) {
-		n, err := literal.DefaultBuilder().Parse(lit)
-		if err != nil {
-			return "", err
-		}
-		return n.ToComparableString(), nil
+	if leftCell.L == nil && leftCell.S == nil {
+		return false, fmt.Errorf("only a literal binding or a string binding (such as an ID binding) should be compared with a literal, but got %v instead", leftCell.String())
 	}
 
-	csEL, err = formatCell(leftCell)
+	lit, err := literal.DefaultBuilder().Parse(e.rS)
 	if err != nil {
 		return false, fmt.Errorf("in comparisonForLiteral.Evaluate got error: %v", err)
 	}
-	csER, err = formatLiteral(e.rS)
+	if leftCell.S != nil && lit.Type() != literal.Text {
+		return false, fmt.Errorf("a string binding can only be compared with a literal of type text, got literal %v instead", lit.String())
+	}
+
+	var (
+		csEL, csER string // comparable string expressions left and right
+	)
```

Comment:
WDYT of:
// comparable string expressions for left and right tokens.
var csEL, csER string

Do note that comments need to have punctuation but error message don't

Reply:
Done.